pkg/activator/handler/enforce_length_handler.go

```diff
@@ -24,10 +26,14 @@ type EnforceMaxContentLengthHandler struct {
 }
 
 func (h *EnforceMaxContentLengthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// If we have a ContentLength, we can fail early
 	if r.ContentLength > h.MaxContentLengthBytes {
 		w.WriteHeader(http.StatusRequestEntityTooLarge)
 		return
 	}
 
+	// Enforce MaxContentLengthBytes
+	r.Body = util.LimitReadCloser(r.Body, h.MaxContentLengthBytes)
```

Comment:
It seems that behavior is different here (error in one case and truncated body in the other). Do we care about it?